### PR TITLE
cargo: bump scm-record to 0.4.0

### LINKED ISSUE #4001: Built-in diff editor doesn't redraw screen correctly in the presence of tabs
## Description

In some files (not sure which), there's a very annoying display bug in the `jj split` UI. Unfolding and folding a section doesn't clear parts of the text, which makes the whole UI kind of unusable in most cases.

Probably best seen in the video below.

## Steps to Reproduce the Problem

1. Have changes in your working tree.
1. `jj split`
1. Unfold and fold a larger file.

## Actual Behavior
 

https://github.com/martinvonz/jj/assets/687230/88ca8a7e-75a0-4f6c-9a3c-ab253396a12c

## Specifications

- Platform: Ubuntu 24.04
- Version: 0.18.0

### LINKED ISSUE #4635: diffedit blank lines on Windows
## Description

Lines are disappearing as you scroll through them in diffedit

## Steps to Reproduce the Problem

1. `jj diffedit`
1. Scroll down with down arrow

## Expected Behavior

Lines should display as normal, just with selection indicated as a background color

## Actual Behavior

image (after unfolding the section and scrolling down twice)

Resizing the window does re-draw it and content is restored to the empty lines, but never to the line currently selected. Could be something to do with rendering the background highlight?

## Specifications

- Platform: Windows
- Version: 0.22.0-67c2ae0a9e3e445f034a456b299e3f37c4dc4b93

Tried it on `cmd.exe`, Windows Terminal and alacritty, powershell and cmd.exe, I don't think this is anything to do with my environment.

### PR DESCRIPTION
**Pull request by farnoy**
Fixes #4001
Fixes #4635 

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Review by arxanas — APPROVED** [review]
Thanks, LGTM! @martinvonz do we have a procedure for remarking on the upgrade in the changelog? (Is it worth remarking? Do we list individual bugs that were fixed, or just say "scm-record was upgraded to v0.4.0" and link to its own release notes?)

**Comment by martinvonz** [comment]
> Thanks, LGTM! @martinvonz do we have a procedure for remarking on the upgrade in the changelog? (Is it worth remarking? Do we list individual bugs that were fixed, or just say "scm-record was upgraded to v0.4.0" and link to its own release notes?)

I don't think we have a procedure for it. Since `scm-record` is so closely integrated that most users seem to think it's simply part of jj, I think it makes sense to at least mention the most important changes in our own changelog. What do you think?

**Comment by arxanas** [comment]
Sounds good to me. I think this would also resolve #4001, and maybe #4502 (we now have a help dialog that just says to click the menu bar items).

**Comment by farnoy** [comment]
FWIW, the original issue that motivated this PR isn't mentioned in scm-record's changelog, so I'm not sure how detailed jj's changelog could be.

IMO, just mentioning a dependency update isn't that useful, unless there's a major change within.

**Comment by arxanas** [comment]
@farnoy I believe it's discussed in this entry:

> (#37): Fixed redraw issues when rendering tabs and other non-printing characters.

since the carriage return is not a printing character. But it's apparent that it's not obvious what to look for. In addition to pointing out the dependency upgrade, we could list each of the individual issues that are resolved in the jj repo in the changelog. I was writing "via X upgrade", e.g.

> git-branchless now supports index version 4 (via libgit2 upgrade).

in the [git-branchless release notes](https://github.com/arxanas/git-branchless/releases/tag/v0.10.0) to point out when issues are resolved via dependency upgrade instead of directly.

**Comment by ilyagr** [comment]
Just try to describe what you think is important about the update, to the degree where it's not much trouble for you. It'll probably be fine, and we can edit it later or suggest fixes here if necessary.

**Comment by farnoy** [comment]
Done

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 * Error on `trunk()` revset resolution is now handled gracefully.
   [#4616](https://github.com/martinvonz/jj/issues/4616)
 
+* Updated the built-in diff editor `scm-record` to version [0.4.0](https://github.com/arxanas/scm-record/releases/tag/v0.4.0), which includes multiple fixes.
```
Comment (ilyagr) [inline CHANGELOG.md:45]: Could you rewrap this text similarly to the other lines?

**Comment by ilyagr** [comment]
Once the wrapping is fixed, I'm planning to merge this in a few hours (to give people time to look, to potentially tell me not to do it, or for Martin to appear and give you the ability to do it yourself).

**Pull request merged**